src/template/page.zig

```diff
@@ -126,26 +141,43 @@ test getOffset {
     // Force into runtime
     var vari: *const []const u8 = undefined;
     const ptr: [*]const u8 = @ptrCast(&vut);
-    vari = @as(*const []const u8, @ptrCast(@alignCast(&ptr[test_5.?])));
+    vari = @as(*const []const u8, @ptrCast(@alignCast(&ptr[test_5])));
     try std.testing.expectEqualStrings("clever girl", vari.*);
 }
 
-fn getChildType(T: type, name: []const u8) type {
+fn baseType(T: type, name: []const u8) type {
     var local: [0xff]u8 = undefined;
     const field = local[0..makeFieldName(name, &local)];
     //return @TypeOf(@FieldType(T, field)); // not in 0.13.0
     for (std.meta.fields(T)) |f| {
         if (eql(u8, f.name, field)) {
-            switch (@typeInfo(f.type)) {
-                .Pointer => |ptr| return ptr.child,
-                .Optional => |opt| return opt.child,
-                .Struct => return f.type,
-                else => unreachable,
+            switch (f.type) {
+                []const u8 => unreachable,
+                ?[]const u8 => unreachable,
+                ?usize => unreachable,
+                else => switch (@typeInfo(f.type)) {
+                    .Pointer => |ptr| return ptr.child,
+                    .Optional => |opt| return opt.child,
+                    .Struct => return f.type,
+                    .Int => return f.type,
```

Comment:
What about float, enum, and union?

Reply:
those are currently impossible type for the template compiler to generate